staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/converter.go

```diff
@@ -162,24 +163,23 @@ func (c *crConverter) ConvertToVersion(in runtime.Object, target runtime.GroupVe
 	fromGVK := in.GetObjectKind().GroupVersionKind()
 	toGVK, ok := target.KindForGroupVersionKinds([]schema.GroupVersionKind{fromGVK})
 	if !ok {
-		// TODO: should this be a typed error?
-		return nil, fmt.Errorf("%v is unstructured and is not suitable for converting to %q", fromGVK.String(), target)
+		return nil, runtime.NewNotRegisteredErrForType(target.Identifier(), reflect.TypeOf(in))
```

Comment:
can you add a unit test that demonstrates the message we'll get here?

The NotRegistered error type seems appropriate, but I think we need to preserve more of the original text (which likely means a new constructor). The first parameter of the calls you're making is the schemeName, which doesn't match what you are passing in, and produces strange messages like `no kind "foo" is registered for version "example.com/v3" in scheme "[example.com/v2]"`